operators/src/pro/cache/shared_cache.rs

```diff
@@ -897,34 +889,44 @@ impl From<VectorLandingQueryEntry> for VectorCacheQueryEntry {
     }
 }
 
+pub trait CacheElement: Sized {
+    type StoredCacheElement: CacheBackendElement<Query = Self::Query>;
+    type Query;
+    type ResultStream: Stream<Item = Result<Self, CacheError>>;
+
+    fn into_stored_element(self) -> Self::StoredCacheElement;
+    fn from_stored_element_ref(stored: &Self::StoredCacheElement) -> Self;
+
+    fn result_stream(
+        stored_data: Arc<Vec<Self::StoredCacheElement>>,
+        query: Self::Query,
+    ) -> Self::ResultStream;
+}
+
 #[async_trait]
 pub trait AsyncCache<C: CacheElement> {
-    async fn query_cache<S: CacheElementSubType<CacheElementType = C>>(
+    async fn query_cache(
```

Comment:
do you still need the CacheElementSubType now?

Reply:
i can remive it but it will result in more code generated by macros.

Reply:
[14e0b15](https://github.com/geo-engine/geoengine/pull/848/commits/14e0b151eac6ce044fb6a2b962b4f2d4ea8540de)